src/components/GenreSelect.tsx

```diff
@@ -0,0 +1,39 @@
+import './GenreSelect.css';
+
+import React from 'react';
+import { useState } from 'react';
+
+interface GenreSelectProps {
+  movieList: string[];
+  selectedMovieName: string;
+  onSelect: Function;
+}
+
+function GenreSelect(props: GenreSelectProps) {
+    const [selectedGenre, setSelectedGenre] = useState<string>(props.selectedMovieName);
+
+    const handleChange = (movieName: string) => {
+        setSelectedGenre(movieName);
```

Comment:
**_Need to improve._** Pay special attention that the setSelectedGenre function is **asynchronous**. So it is **not** set a change immediately.
For this reason, the next function will most likely get an unexpected value.

https://github.com/Kinoler/ReactJsCource/assets/47754466/8e35b927-c8c6-43d3-be24-e0b9e26d50c9

Reply:
How to fix it:
1. Store information about the state of the selected genre in the parent component. (Recommended because you haven't viewed the Hooks module yet)

That is, you need to transfer the useState to the App component. Next, Add a selectedGenre to the props as selectedMovieName and setSelectedGenre as onSelect to your component.
![2023-09-01_18h14_15](https://github.com/Kinoler/ReactJsCource/assets/47754466/95a85a61-d33b-41e2-8721-e6c77e5b0f01)

2. Add useEffect hook.
https://react.dev/reference/react/useEffect
![image](https://github.com/Kinoler/ReactJsCource/assets/47754466/d24e6ff5-b19b-4322-a27b-0e69aa8696b6)